catalogue/webapp/utils/dates.ts

```diff
@@ -52,17 +52,32 @@ export function convertDayNumberToDay(dayNumber: DayNumber): Day {
   }
 }
 
-export function findNextRegularOpenDay(
+export function findNextPickUpDay(
   date: Moment,
   regularClosedDays: DayNumber[]
 ): Moment | undefined {
   if (regularClosedDays.length === 7) {
     // All days are closed, so we'll never be able to find a non closed day.
     return undefined;
   }
+
+  // If the library is closed on this day, we want to set the pick-up day to be
+  // the next open day plus one, so that e.g. Monday morning isn't a scramble
+  // for library staff handling the weekend's requests. Since this function
+  // calls itself recursively, we add one day if we're closed today and
+  // tomorrow, but add two days if we're closed today and open tomorrow.
+
+  const nextDay = date.clone().add(1, 'days');
   const isClosed = regularClosedDays.includes(date.day() as DayNumber);
+  const isLastClosedDay = !regularClosedDays.includes(
+    nextDay.day() as DayNumber
+  );
+
   if (isClosed) {
-    return findNextRegularOpenDay(date.add(1, 'day'), regularClosedDays);
+    return findNextPickUpDay(
+      date.add(isLastClosedDay ? 2 : 1, 'day'),
+      regularClosedDays
+    );
   } else {
     return date;
   }
```

Comment:
I think your comment hints at a clearer naming here:

```typescript
const isOpenTomorrow = !regularClosedDays.includes(nextDay.day() as DayNumber);

if (isClosed && isOpenTomorrow) {
  return findNextPickUpDay(date.add(2, 'day'), regularClosedDays));
} else if (isClosed) {
  return findNextPickUpDay(date.add(1, 'day'), regularClosedDays));
} else {
  return date;
}
```

Reply:
I went back and forth on `isOpenTomorrow` and whether it was better or worse than something like `isOpenNextDay` (since 'tomorrow' is relative to 'today' while 'next day' is relative to the date that the function is being called with. Then plumped for `isLastClosedDay`. But maybe the intent of `isOpenTomorrow` is clearer and I'm overthinking it.

Reply:
I guess if we go with `isOpenTomorrow`, it would make sense for the `nextDay` variable to be renamed to `tomorrow`, too. And then going the full hog, renaming `date` to `today`

Reply:
Ah, no, you're right, I think `isOpenNextDay` is correct – because `date` might not be `today` if we're calling this recursively?

Reply:
Yeah, I think I'm coming down in favour of

```
 const nextDay = date.clone().add(1, 'days');
  const isClosedThisDay = regularClosedDays.includes(date.day() as DayNumber);
  const isOpenNextDay = !regularClosedDays.includes(nextDay.day() as DayNumber);

  if (isClosedThisDay && isOpenNextDay) {
    return findNextPickUpDay(date.add(2, 'day'), regularClosedDays);
  } else if (isClosedThisDay) {
    return findNextPickUpDay(date.add(1, 'day'), regularClosedDays);
  } else {
    return date;
  }
```

Reply:
Yeah, that looks better.